### PR TITLE
Prevent attempts to delete from federated storage when removing a service account user

### LINKED ISSUE #12296: Deadlock when calling removeUser for Service Account Linked Users
### Describe the bug

I'm with the IT User team at Red Hat. We are working on introducing an API which will allow dynamic creation / deletion of service accounts in one of our realms. The realm is configured with federated user storage. We are frequently noticing the following deadlock during deletion of service accounts:
```
Uncaught server error: javax.persistence.OptimisticLockException: org.hibernate.exception.LockAcquisitionException: could not execute statement
	at org.hibernate@5.3.23.Final-redhat-00001//org.hibernate.internal.ExceptionConverterImpl.wrapLockException(ExceptionConverterImpl.java:277)
	at org.hibernate@5.3.23.Final-redhat-00001//org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:98)
	at org.hibernate@5.3.23.Final-redhat-00001//org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181)
	at org.hibernate@5.3.23.Final-redhat-00001//org.hibernate.query.internal.AbstractProducedQuery.executeUpdate(AbstractProducedQuery.java:1654)
	at org.keycloak.keycloak-model-jpa@15.0.4.redhat-00001//org.keycloak.storage.jpa.JpaUserFederatedStorageProvider.preRemove(JpaUserFederatedStorageProvider.java:826)
	at org.keycloak.keycloak-services@15.0.4.redhat-00001//org.keycloak.storage.UserStorageManager.removeUser(UserStorageManager.java:252)
	at org.keycloak.keycloak-model-infinispan@15.0.4.redhat-00001//org.keycloak.models.cache.infinispan.UserCacheSession.removeUser(UserCacheSession.java:802)
	at org.keycloak.keycloak-model-infinispan@15.0.4.redhat-00001//org.keycloak.models.cache.infinispan.RealmCacheSession.removeClient(RealmCacheSession.java:601)
	at org.keycloak.keycloak-model-infinispan@15.0.4.redhat-00001//org.keycloak.models.cache.infinispan.RealmAdapter.removeClient(RealmAdapter.java:809)
	at org.keycloak.keycloak-services@15.0.4.redhat-00001//org.keycloak.services.managers.ClientManager.removeClient(ClientManager.java:99)
```

It is failing on the named queries found in `JpaUserFederatedStorageProvider::preRemove`. Called by `UserStorageManager::removeUser`.

It seems these queries are responsible for cleaning up federated user relations during deletion of a user. I think for the internally created service account user, these may not be necessary. And could help prevent deadlocks and improve performance by not querying and locking against these contentious tables.

Let me know what you think.

### Version

RH SSO 7.5

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by patrickjennings**
Do not attempt to delete from federated user tables, when deleting a service account linked user.

These are highly contentious tables in our environments. Deleting a service account user is causing deadlocks in this code path.

I don't think it is possible for these service account users to have links to brokers / federate storage providers. But please let me know if there are any use cases where that might be.

closes #12296

### CODE REVIEW

**Comment by alechenninger** [comment]
@stianst @hmlnarik can we get some eyes on this one?

**Comment by hmlnarik** [comment]
Thank you for the fix. This needs to wait until #11933 is merged, as there are conflicts between these two PRs.

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**
